File: catalogo/src/main/java/cinerama/catalogo/models/Horario.java
package cinerama.catalogo.models;
import java.time.LocalDate;
import java.time.LocalTime;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "horarios") 
@Data 
public class Horario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "fecha", nullable = false)
    private LocalDate fecha;

    @Column(name = "hora_inicio", nullable = false)
    private LocalTime horaInicio;

    @Column(name = "precio", nullable = false)
    private Double precio;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pelicula_id", nullable = false)
    private Pelicula pelicula;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sala_id", nullable = false)
    private Sala sala;
}

File: frontend/src/app/core/models/horario.ts
export interface Horario {
  id: number;
  fecha: string;
  horaInicio: string; // Llegará en formato "HH:mm:ss"
  precio: number;
  peliculaId: number;
  tituloPelicula: string;
  salaId: number;
  numeroSala: number;
  cineId: number;
  nombreCine: string;
}